test/file-transfer-handler.test.js

```diff
@@ -0,0 +1,260 @@
+        should(uploadResult.getFileUploadResults().length).be.exactly(1);
+        await setTimeoutPromise(10);
+        fileUploadResult.stopTimer();
+        should(fileUploadResult.getTotalUploadTime()).be.ok();
+        should(fileUploadResult.getFileName()).be.exactly(fileInfo.getFileName());
+
+        // subsequent calls should not add more transferStarted items
+        started = await handler.partTransferStarted(uploadResult, initResponseFileInfo);
+        should(started).be.ok();
+        should(transferStarted.length).be.exactly(1);
+    });
+
+    it('test part transfer started cancelled', async function() {
+        const uploadResult = new UploadResult(options, uploadOptions);
+
+        httpClient.cancel(FILE_PATH);
+
+        const started = await handler.partTransferStarted(uploadResult, initResponseFileInfo);
+        should(started).not.be.ok();
+        should(transferStarted.length).be.exactly(0);
+        should(uploadResult.getFileUploadResults().length).be.exactly(1);
+    });
+
+    it('test part transfer started cancelled after started', async function() {
+        const uploadResult = new UploadResult(options, uploadOptions);
+
+        const started = await handler.partTransferStarted(uploadResult, initResponseFileInfo);
+        should(started).be.ok();
+
+        httpClient.cancel(FILE_PATH);
+
+        should(await handler.partTransferStarted(uploadResult, initResponseFileInfo)).not.be.ok();
+    });
+
+    it('test part transfer started not succcessful', async function() {
+        const uploadResult = new UploadResult(options, uploadOptions);
+
+        let startCount = 0;
+        handler._doFileTransferStarted = async(fileUploadResult) => {
+            fileUploadResult.setCompleteError('there was a problem');
+            startCount++;
+        };
+
+        const started = await handler.partTransferStarted(uploadResult, initResponseFileInfo);
+        should(started).not.be.ok();
+        should(startCount).be.exactly(1);
+
+        should(await handler.partTransferStarted(uploadResult, initResponseFileInfo)).not.be.ok();
+        should(startCount).be.exactly(1);
+    });
+
+    it('test part transfer progress', async function() {
+        await handler.partTransferProgress(initResponseFileInfo, 100);
+        should(transferProgress.length).be.exactly(0);
+
+        await handler.partTransferStarted(new UploadResult(options, uploadOptions), initResponseFileInfo);
+        await handler.partTransferProgress(initResponseFileInfo, 100);
+        should(transferProgress.length).be.exactly(1);
+
+        const { initResponseFileInfo: progressInfo, fileUploadResult, transferred } = transferProgress[0];
+        should(progressInfo.getFileName()).be.exactly(FILE_NAME);
+        should(fileUploadResult.getFileName()).be.exactly(progressInfo.getFileName());
+        should(transferred).be.exactly(100);
+
+        await handler.partTransferProgress(initResponseFileInfo, 100);
```

Comment:
You could also run those 3 lines in "parallel", aka `await Promise.all(<<the 3 calls here as array>>).

Reply:
Nice! Thanks for the suggestion. I made an update here and will keep `Promise.all()` in my toolbelt 👍